sdk/core/azure-core/tests/test_user_agent_policy.py

```diff
@@ -0,0 +1,22 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Tests for the user agent policy."""
+from azure.core.pipeline.policies import UserAgentPolicy
+from azure.core.pipeline.transport import HttpRequest
+from azure.core.pipeline import PipelineRequest, PipelineContext
+
+def test_user_agent_policy():
+    user_agent = UserAgentPolicy(base_user_agent='foo')
+    assert user_agent._user_agent == 'foo'
+
+    user_agent = UserAgentPolicy(base_user_agent='foo', user_agent='bar')
+    assert user_agent._user_agent == 'bar foo'
+
+    request = HttpRequest('GET', 'http://127.0.0.1/')
+    pipeline_request = PipelineRequest(request, PipelineContext(None))
+
+    pipeline_request.context.options['user_agent'] = 'xyz'
+    user_agent.on_request(pipeline_request)
+    assert request.headers['User-Agent'] == 'xyz bar foo'
```

Comment:
I think you will need to explicitly clear `os.environ[self._ENV_ADDITIONAL_USER_AGENT]` for this to pass as-is (and then a test with it set explicitly would also be advised, if there is not one elsewhere already)

Reply:
Yes. Updated the tests.

Reply:
I didn't even know about `user_agent_use_env` that just suggests more possible tests:

* user_agent_use_env=True, env var set
* user_agent_use_env=True, env var not set
* user_agent_use_env=False, env var set
* user_agent_use_env=False, env var not set